lib/fairy/fairy.c

```diff
@@ -258,13 +259,61 @@ void Fairy_InitFile(FairyFileInfo* fileInfo, FILE* file) {
 
                     /* Ignore the leading "." */
                     if (strcmp(&shstrtab[currentSection.sh_name + 1], "text") == 0) {
-                        fileInfo->progBitsSizes[FAIRY_SECTION_TEXT] += ALIGN(currentSection.sh_size, 0x10);
+                        if (gUseElfAlignment) {
+                            size_t alignedSize;
+
+                            /* Ensure the next file will start at its correct alignment */
+                            fileInfo->progBitsSizes[FAIRY_SECTION_TEXT] = ALIGN(fileInfo->progBitsSizes[FAIRY_SECTION_TEXT], currentSection.sh_addralign);
+
+                            alignedSize = ALIGN(currentSection.sh_size, currentSection.sh_addralign);
+
+                            FAIRY_DEBUG_PRINTF("%s section alignment: 0x%X\n", &shstrtab[currentSection.sh_name + 1], currentSection.sh_addralign);
```

Comment:
I think you don't want the ` + 1`s here, unless you want to leave the dots off?

Reply:
The already existing debug printfs don't have the dot, so I left the +1 for consistency